routing/dht/handlers.go

```diff
@@ -226,5 +226,5 @@ func (dht *IpfsDHT) handleAddProvider(ctx context.Context, p peer.ID, pmes *pb.M
 		dht.providers.AddProvider(ctx, key, p)
 	}
 
-	return pmes, nil // send back same msg as confirmation.
```

Comment:
agreed on that. but some dht protocols (particularly if we start doing cryptographic proofs on things) may call for other constructions too. in any case, not very important. either way